modules/openapi-generator/src/main/resources/go-experimental/model.mustache

```diff
@@ -111,26 +123,31 @@ func (o *{{classname}}) Set{{name}}(v {{dataType}}) {
 {{^required}}
 // Get{{name}} returns the {{name}} field value if set, zero value otherwise.
 func (o *{{classname}}) Get{{name}}() {{dataType}} {
-	if o == nil || o.{{name}} == nil {
+	if o == nil {{^isNullable}}|| o.{{name}} == nil{{/isNullable}} {
 		var ret {{dataType}}
 		return ret
 	}
-	return *o.{{name}}
+	return {{^isNullable}}*{{/isNullable}}o.{{name}}
 }
 
 // Get{{name}}Ok returns a tuple with the {{name}} field value if set, zero value otherwise
 // and a boolean to check if the value has been set.
 func (o *{{classname}}) Get{{name}}Ok() ({{dataType}}, bool) {
-	if o == nil || o.{{name}} == nil {
+	if o == nil {{^isNullable}}|| o.{{name}} == nil{{/isNullable}} {
 		var ret {{dataType}}
 		return ret, false
 	}
+{{#isNullable}}
+	return o.{{name}}, {{#isContainer}}o.{{name}} == nil{{/isContainer}}{{^isContainer}}o.{{name}}.IsSet(){{/isContainer}}
```

Comment:
Since we return if the value inside the nullable is set, why don't we return the value inside the nullable too directly ? Might simplify usage of nullables ?
Haven't thought about it deeply, just throwing that out there. Let me know what you think

Reply:
I think it would then be problematic what `Get{{name}}` should return - because it would return null both if there's explicit null and if there's not, so it wouldn't make much sense to keep that method for nullables, which would lead to inconsistency in the API of the class. Also, getting the whole nullable structure directly lets you call e.g. `Unset()` on it... I do see your point, but I think this is a matter of preference, YMMV :)

Reply:
Hmm, but it's true that doing this would make it somewhat easier to add implementation of nullables for container types in the future. Let me try to experiment with the idea a bit.